cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -912,19 +912,191 @@ def create_image(self, name):
         raise NotImplementedError(
             'To be implemented after GCEVolumeService.')
 
+    @property
+    def _existing_target_instance(self):
+        self_url = self._provider.parse_url(self._gce_instance['selfLink'])
+        try:
+            response = (self._provider.gce_compute
+                                      .targetInstances()
+                                      .list(project=self._provider.project_name,
+                                            zone=self_url.parameters['zone'])
+                                      .execute())
+            if 'items' not in response:
+                return None
+            for target_instance in response['items']:
+                url = self._provider.parse_url(target_instance['instance'])
+                if url.parameters['instance'] == self.name:
+                    return target_instance
+        except Exception as e:
+            cb.log.warning('Exception while listing target instances: %s', e)
+        return None
+
+    @property
+    def _target_instance(self):
+        existing_target_instance = self._existing_target_instance
+        if existing_target_instance:
+            return existing_target_instance
+
+        # No targetInstance exists for this instance. Create one.
+        self_url = self._provider.parse_url(self._gce_instance['selfLink'])
+        body = {'name': 'target-instance-{0}'.format(uuid.uuid4()),
+                'instance': self._gce_instance['selfLink']}
+        try:
+            response = (self._provider.gce_compute
+                                      .targetInstances()
+                                      .insert(
+                                          project=self._provider.project_name,
```

Comment:
As explained above, if you are OK with using `self._provider.project_name` I prefer to be consistent. I think we have used `self._provider.project_name` in all API calls.